event_buffer.go

```diff
@@ -0,0 +1,71 @@
+package statful
+
+import (
+	"bytes"
+	"encoding/json"
+	"sync"
+)
+
+type eventBuffer struct {
+	buffer     []Event
+	dryRun     bool
+	eventCount int
+	flushSize  int
+	mu         sync.Mutex
+
+	Logger Logger
+	Sender Sender
+}
+
+func (e *eventBuffer) Event(event Event) {
+	e.mu.Lock()
+
+	e.buffer = append(e.buffer, event)
+	e.eventCount++
+
+	e.mu.Unlock()
```

Comment:
Can we defer this right after locking? It's a good practice when possible to avoid forgetting unlocking and has the side benefit of making the code more readable and clearer.